### PR TITLE
Transpile with Babel to ES5

### PR DESCRIPTION
**Pull request by 40thieves**
### Description

Uses Babel to transpile the module to ES5, allowing it to be used in (older) browsers.

### Review

Based on https://github.com/overleaf/o-error/pull/15, but https://github.com/overleaf/o-error/pull/15#issuecomment-762811597 points out that `tsc`'s output doesn't appear to be the same stack trace.

**With Babel**

```
> const OError = require('./dist')
> new OError('hi')
OError: hi
    at repl:1:1
```

(Interestingly, I'm not seeing the `Error` prefix that @jdleesmiller has. Possibly minor version mismatch, I'm on v12.18.2)

The tests also pass when pointed at `dist`. I did wonder if we should be testing with the source or dist files, but decided to leave as is for now.

I configured preset-env to use the same browserlist config as we use in the frontend. I can switch this to the default if wanted, but since we support IE11 I think there wouldn't be much difference.

#### Manual tests

TODO: I'd like to try pulling this into a copy of web just to make sure.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,15 +12,17 @@
...
+    "dist/index.d.ts"
   ],
   "scripts": {
-    "build": "npm run --silent typecheck && npm run --silent test && npm run --silent declaration:build && npm run --silent update-readme",
-    "declaration:build": "rm -f index.d.ts && tsc --allowJs --declaration --emitDeclarationOnly --moduleResolution node --target ES6 index.js",
-    "declaration:check": "git diff --exit-code -- index.d.ts",
+    "clean": "rm -rf dist",
+    "build": "npm run --silent clean && npm run --silent typecheck && npm run --silent test && npm run --silent build:compile && npm run --silent declaration:build && npm run --silent update-readme",
+    "build:compile": "babel index.js --out-dir dist",
+    "declaration:build": "rm -f dist/index.d.ts && tsc --allowJs --declaration --emitDeclarationOnly --moduleResolution node --target ES6 index.js",
```
Comment (aeaton-overleaf) [inline package.json:24]: tsc needs `--outDir dist`, so it puts `index.d.ts` in the right place

Comment (40thieves) [reply package.json]: Thanks fixed in 694dfa2

**Comment by 40thieves** [comment]
I'm going to move this to review. I did try to test in web with a local branch, but managing all the npm/containers was taking a long time to debug.

Assuming the review is ok, we can cut a beta release to test with web much more simply.

**Review by jdleesmiller — APPROVED** [review]
👍 looks good.

Suggest we put it on master and pull it in as a tarball to test initially.

**Comment by 40thieves** [comment]
Did we decide which version to bump to? Is this a major? (I would lean towards no, as there should be no breaking changes, but it is the safer option).

**Comment by jdleesmiller** [comment]
I'm leaning toward a minor, too.

**Pull request merged**